routing/payment_session.go

```diff
@@ -182,6 +182,26 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 			return nil, err
 		}
 
+		// Pre-check whether the total balance is sufficient for the
+		// payment. Otherwise pathfinding and splitting won't ever
+		// result in a successful payment, but small shards may be held
+		// by the recipient until mpp timeout.
+		var totalBandwidth lnwire.MilliSatoshi
+		for chanID, bandwidth := range bandwidthHints {
+			// Only count outgoing channel if there is a
+			// restriction.
+			if p.payment.OutgoingChannelID != nil &&
+				*p.payment.OutgoingChannelID != chanID {
+
+				continue
+			}
+
+			totalBandwidth += bandwidth
+		}
+		if totalBandwidth < maxAmt {
+			return nil, errNoPathFound
```

Comment:
Should we reuse insufficient local balance error?

Reply:
Yes, this may be the moment. I brought it back.